### PR TITLE
Fix profile menu config click action issues

### LINKED ISSUE #1561: [Quest/eCBIS] - Fix issues affecting Remove family functionality
- [ ] Ensure `Remove family` profile menu action removes family. Currently clicking `Remove Family` button in the questionnaire activity causes a crash.
- [ ] Ensure families that are not active are filtered out of the house hold register
- [ ] Refactor `remove family` and `remove family member` functionality to use `QuestionnaireActivity` and `QuestionnaireViewModel` classes
- [ ] Delete all classes in `family.remove` package

### PR DESCRIPTION
**Pull request by Rkareko**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1561

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt [hunk 1]
```diff
@@ -217,10 +217,11 @@ constructor(open val fhirEngine: FhirEngine, open val dispatcherProvider: Dispat
     }
   }
 
-  /** Remove member of a group using the provided [patientId] */
-  suspend fun removeGroupMember(patientId: String, groupId: String?) {
+  /** Remove member of a group using the provided [memberId] */
+  suspend fun removeGroupMember(memberId: String, groupId: String?) {
     // TODO refactor to work with any resource type
-    loadResource<Patient>(patientId)?.let { patient ->
+    // TODO provide resourceType as param
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt:223]: can we combine this todo w/the one above? reads like they have the same meaning

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt [hunk 2]
```diff
@@ -294,3 +294,9 @@ fun Composition.retrieveCompositionSections(): List<Composition.SectionComponent
 fun String.resourceClassType(): Class<out Resource> {
   return Class.forName("org.hl7.fhir.r4.model.$this") as Class<out Resource>
 }
+/**
+ * A function that returns the logical id from the result of a fhirpath that extracts the Resource
+ * id e.g running this fhirpath expression "Group.id" returns an id with the following format
+ * [Group/0acda8c9-3fa3-40ae-abcd-7d1fba7098b4/_history/2]
+ */
+fun String.logicalIdFromFhirPathExtractedId() = this.split("/")?.get(1)
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt:302]: isn't it possible for this to throw an error if the split on "/" returns only 1 or fewer items? can we guard against that if so

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt [hunk 3]
```diff
@@ -113,6 +110,11 @@ constructor(
     )
   }
 
+  // TODO this setting should be loaded from a configuration
+  var isDeactivateMembers = false
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:114]: can you describe what this var controls? From the name that is not unclear to me

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterViewModel.kt [hunk 4]
```diff
@@ -138,8 +138,8 @@ constructor(
       is RegisterEvent.RegisterNewClient ->
         event.context.launchQuestionnaire<QuestionnaireActivity>(
           // TODO use appropriate property from the register configuration
-          "provide-questionnaire-id"
-        )
+          // "provide-questionnaire-id"
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/register/RegisterViewModel.kt:141]: do we still need this event, how is it deciding what to launch now?

**Review by pld — DISMISSED** [review]

**Comment by ellykits** [comment]
Merging this. Will create another PR with the updated application configs to fix addMember and removeFamily actions on family profile.

**Pull request merged**